xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -1560,3 +1578,58 @@ DVDNavVideoStreamInfo CDVDInputStreamNavigator::GetVideoStreamInfo()
 
   return info;
 }
+
+int dvd_inputstreamnavigator_cb_seek(void * p_stream, uint64_t i_pos)
+{
+  CDVDInputStreamFile *lpstream = (CDVDInputStreamFile*)p_stream;
+  if (lpstream->Seek(i_pos, 0) >= 0)
+    return 0;
+  else
+    return -1;
+}
+
+int dvd_inputstreamnavigator_cb_read(void * p_stream, void * buffer, int i_read)
+{
+  CDVDInputStreamFile *lpstream = (CDVDInputStreamFile*)p_stream;
+  return lpstream->Read((uint8_t *)buffer, i_read);
+}
+
+int dvd_inputstreamnavigator_cb_readv(void * p_stream, void * p_iovec, int i_blocks)
+{
+  // NOTE/TODO: this vectored read callback somehow doesn't seem to be called by libdvdnav.
+  // Therefore, the code below isn't really tested, but inspired from the libc_readv code for Win32 in libdvdcss (device.c:713). 
+  CDVDInputStreamFile *lpstream = (CDVDInputStreamFile*)p_stream;
+  const struct iovec* lpiovec = (const struct iovec*)p_iovec;
+
+  int i_index, i_len, i_total = 0;
+  unsigned char *p_base;
+  int i_bytes;
+
+  for (i_index = i_blocks; i_index; i_index--, lpiovec++)
```

Comment:
the code is correct, it's a vector of iovec structs, so the pointer increments with the struct size. The code was copied from libdvdcss.